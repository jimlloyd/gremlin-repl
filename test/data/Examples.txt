// See https://github.com/inolen/gremlin-node/blob/master/README.md#examples

// Note: we use  in several places below simply to guarantee determinstic 
// results, which is necessary for this to be a stable unit test.
// Unfortunately it adds complexity to the examples. In all cases, the .order() step can be ignored
// without invalidating the example.

// Example 1: Basic Transforms
g.V('name', 'marko').out()
// g.V({name: 'marko'}).out();
// g.v(1, 4).out('knows', 'created').in();
// g.v([1, 4]).out(['knows', 'created']).in();

// Example 2: [i]
g.V().order('{it -> it.a.id <=> it.b.id}').index(0).property('name');

// Example 3: [i..j]
g.V().order('{it -> it.a.id <=> it.b.id}').range(0,1)    // Note, 0 and 1 are not the id's '0' and '1'.
g.V().order('{it -> it.a.id <=> it.b.id}').range(0,1).property('name');

// Example 4: has
g.E().has('id', T.eq, '9')
// has with comparison to float. Must use g.java.newFloat().
// Doesn't seem to be possible to automatically cast from javascript double precision
// to java float precision.
g.E().has('weight', T.gt, g.java.newFloat(0.5))
g.E().has('weight', T.lt, g.java.newFloat(0.5))
g.E().has('weight', T.eq, g.java.newFloat(0.5))
g.E().has('weight', T.neq, g.java.newFloat(0.5))
g.E().has('weight', T.gte, g.java.newFloat(0.5))
g.E().has('weight', T.lte, g.java.newFloat(0.5))

g.E().has('weight', T.gt, g.java.newFloat(0.5)).outV().transform('{it -> [it.id, it.age]}')

// Example 5: and & or
g.V().and(g._().both('knows'), g._().both('created'))
g.V('id', '1').outE().or(g._().has('id', T.eq, '9'), g._().has('weight', T.gt, g.java.newFloat(0.6)))

// g.V().out().groupBy('{it.name}{it.in}{it.unique().findAll{i -> i.age > 30}.name}').cap();
// g.V().retain([g.v(1), g.v(2), g.v(3)]);

var m = new g.HashMap();
g.V().order('{it -> it.a.id <=> it.b.id}').groupBy(m, '{it -> it}', '{it -> it.out}').iterate()
// m

var x = new g.ArrayList()
g.V('id', '1').out().order('{it -> it.a.id <=> it.b.id}').aggregate(x).out().retain(x)
x


