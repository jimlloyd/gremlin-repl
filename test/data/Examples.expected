node > // See https://github.com/inolen/gremlin-node/blob/master/README.md#examples
node > 
node > // Example 1: Basic Transforms
node > g.V('name', 'marko').out();
[ { age: 27, name: 'vadas', _id: '2', _type: 'vertex' },
  { age: 32, name: 'josh', _id: '4', _type: 'vertex' },
  { name: 'lop', lang: 'java', _id: '3', _type: 'vertex' } ]
node > // g.V({name: 'marko'}).out();
node > // g.v(1, 4).out('knows', 'created').in();
node > // g.v([1, 4]).out(['knows', 'created']).in();
node > 
node > // Example 2: [i]
node > g.V().index(0).property('name');
[ 'lop' ]
node > 
node > // Example 3: [i..j]
node > g.V().range(0,1)    // Note, 0 and 1 are not the id's '0' and '1'.
[ { name: 'lop', lang: 'java', _id: '3', _type: 'vertex' },
  { age: 27, name: 'vadas', _id: '2', _type: 'vertex' } ]
node > g.V().range(0,1).property('name');
[ 'lop', 'vadas' ]
node > 
node > // Example 4: has
node > g.E().has('id', T.eq, '9')
[ { weight: 0.4,
    _id: '9',
    _type: 'edge',
    _outV: '1',
    _inV: '3',
    _label: 'created' } ]
node > // has with comparison to float. Must use g.java.newFloat().
node > // Doesn't seem to be possible to automatically cast from javascript double precision
node > // to java float precision.
node > g.E().has('weight', T.gt, g.java.newFloat(0.5)).map()
[ { weight: 1 }, { weight: 1 } ]
node > g.E().has('weight', T.lt, g.java.newFloat(0.5)).map()
[ { weight: 0.4 }, { weight: 0.4 }, { weight: 0.2 } ]
node > g.E().has('weight', T.eq, g.java.newFloat(0.5)).map()
[ { weight: 0.5 } ]
node > g.E().has('weight', T.neq, g.java.newFloat(0.5)).map()
[ { weight: 1 },
  { weight: 0.4 },
  { weight: 1 },
  { weight: 0.4 },
  { weight: 0.2 } ]
node > g.E().has('weight', T.gte, g.java.newFloat(0.5)).map()
[ { weight: 1 }, { weight: 0.5 }, { weight: 1 } ]
node > g.E().has('weight', T.lte, g.java.newFloat(0.5)).map()
[ { weight: 0.5 },
  { weight: 0.4 },
  { weight: 0.4 },
  { weight: 0.2 } ]
node > 
node > g.E().has('weight', T.gt, g.java.newFloat(0.5)).outV().transform('{it -> [it.id, it.age]}')
[ [ '4', 32 ], [ '1', 29 ] ]
node > 
node > // Example 5: and & or
node > g.V().and(g._().both('knows'), g._().both('created')); // in mine, not in inolen's
[ { age: 29, name: 'marko', _id: '1', _type: 'vertex' },
  { age: 32, name: 'josh', _id: '4', _type: 'vertex' } ]
node > g.V('id', '1').outE().or(g._().has('id', T.eq, '9'), g._().has('weight', T.gt, g.java.newFloat(0.6)));
[ { weight: 1,
    _id: '8',
    _type: 'edge',
    _outV: '1',
    _inV: '4',
    _label: 'knows' },
  { weight: 0.4,
    _id: '9',
    _type: 'edge',
    _outV: '1',
    _inV: '3',
    _label: 'created' } ]
node > 
node > // g.V().out().groupBy('{it.name}{it.in}{it.unique().findAll{i -> i.age > 30}.name}').cap();
node > // g.V().retain([g.v(1), g.v(2), g.v(3)]);
node > 
node > var map = new g.HashMap();
node > //g.V().groupBy(map, '{it}{it.out}').iterate();
node > 
node > var x = new g.ArrayList()
node > g.V('id', '1').out().aggregate(x).out().retain(x)
[ { name: 'lop', lang: 'java', _id: '3', _type: 'vertex' } ]
node > 
node > 